File: src/gestorAplicacion/clasesHerencia/TecnoVans.java
package gestorAplicacion.clasesHerencia;

import java.util.ArrayList;
import gestorAplicacion.clasesPrincipales.*;

import gestorAplicacion.clasesPrincipales.Bus;
import gestorAplicacion.clasesPrincipales.Conductor;

public class TecnoVans extends Bus{
	int codigo;
	private final int precioKm=800;//precio por km
	private final int capacidad=22;
	public static ArrayList<TecnoVans> tecnosV = new ArrayList<TecnoVans>();

	
	//constructor
		public TecnoVans(String placa, Conductor conductor, ArrayList<String> sillas) {
			super(placa,conductor,sillas);
			this.codigo=200;
			TecnoVans.tecnosV.add(this);
		}
	//get&set capacidad
		public int getCapacidad() {
			return capacidad;
		}

		public int getPrecioKm() {
			return precioKm;
		}
		//get&set codigo
		public int getCodigo() { //desde aqui agregue 
			return codigo;
		}

		public void setCodigo(int codigo) {
			this.codigo = codigo;
		}
		@Override
		public void sillasDisponibles() {
			for(Tiquete tiquete : Viaje.getTiquetesTodos()){
				if(tiquete.getEstado() == false){
					System.out.println("la silla: "+tiquete.getSillaTiquete()+"esta disponible");
					}
				}
			}
		public String sillaTiquete(Tiquete tiquete) {
			return tiquete.getSillaTiquete();
		}
}

File: Python/TecnoVans.py
from Bus import Bus
from Viaje import Viaje

class TecnoVans(Bus):

    def __init__(self,placa,conductor,sillas,precioKm=800,codigo=200,capacidad=22):
        tecnosV = []
        super().__init__(placa,conductor,sillas)
        TecnoVans.tecnosV.append(self)

    #get&set capacidad
    def getCapacidad(self):
        return self._capacidad

    #get&set PrecioKm
    def getPrecioKm(self):
        return self._precioKm

    #get&set codigo
    def getCodigo(self):
        return self.codigo

    #MetodoSillasDispo
    def sillasDisponibles(self):
        for tiquete in Viaje.getTiquetesTodos():
            if tiquete.getEstado() == False:
                print("la silla: "+tiquete.getSillaTiquete()+"esta disponible")
            
    def sillaTiquete(self, tiquete):
        return tiquete.getSillaTiquete()